extensions/windows-event-log/ConsumeWindowsEventLog.cpp

```diff
@@ -271,19 +280,88 @@ void ConsumeWindowsEventLog::onTrigger(const std::shared_ptr<core::ProcessContex
     return;
   }
 
-  const auto flowFileCount = processQueue(session);
+  struct TimeDiff {
+    auto operator()() const {
+      return std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::steady_clock::now() - time_).count();
+    }
+    const decltype(std::chrono::steady_clock::now()) time_ = std::chrono::steady_clock::now();
+  };
+
+  const auto commitAndSaveBookmark = [&] (const std::wstring& bookmarkXml) {
+    const TimeDiff timeDiff;
+    session->commit();
+    logger_->log_debug("processQueue commit took %" PRIu64 " ms", (uint64_t)timeDiff());
```

Comment:
Please don't use C-style casts! My suggestion about the explicit return value would make it redundant in this case.

http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es48-avoid-casts
http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es49-if-you-must-use-a-cast-use-a-named-cast (note: here, we don't need a cast)

Reply:
We need cast because of `narrowing conversion` error.

Reply:
Shouldn't be the case since you've changed the function to return `int64_t`